### PR TITLE
feat(Data Push):  have a single delivery format of the push notification metadata. 🧾 

### PR DESCRIPTION
**Pull request by Santi92**
## Context

It happens that to read the push metadata, the structure was different for each platform.

## Solution 

In this will PR  contains a push read to formalize the structure before sending the clients

This will PR

✅  add a `sanitizeMap` method for formalize the structure.
✅. Unit test for each platforms. 🤓 

## No Changes in iOS

> In iOS yeas sends the data correctly 

## Breaking changes 🔥 

> This functionality breaks the metadata reading that is currently being done in Android 🔥 

## Current behavior 

Android 

image (41)
<details>
  <summary>Show map complete!</summary>

  ```dart
      {
        title: HiPush,
        body: test,
        additionalData: {
          keyMap: {
            "keyMap2": "value3",
            "keyMap1": "value2",
            "keyMap4": "value4"
          },
          keyMapChild: {
            "keyMapChild1": "value2",
            "keyMapChild2": "value3",
            "keyMapChild3": {
              "keyMapChild32": "value3",
              "keyMapChild31": "value2",
              "keyMapChild34": "value4"
            }
          },
          itbl: {
            "defaultAction": {
              "type": "openApp"
            },
            "isGhostPush": false,
            "messageId": "12",
            "actionButtons": [
              
            ],
            "templateId": 12
          },
          keyNumber: 1,
          body: test,
          title: HiPush,
          actionIdentifier: default,
          key: value1
        }
      }
```
</details>

## Expected behavior 

image (42)

<details>
  <summary>Show map complete!</summary>
 
   ```dart
     {
      title: HiPush,
      body: test,
      additionalData: {
        keyMap: {
          keyMap2: value3,
          keyMap1: value2,
          keyMap4: value4
        },
        keyMapChild: {
          keyMapChild1: value2,
          keyMapChild2: value3,
          keyMapChild3: {
            keyMapChild32: value3,
            keyMapChild31: value2,
            keyMapChild34: value4
          }
        },
        itbl: {
          defaultAction: {
            type: openApp
          },
          isGhostPush: false,
          messageId: 123,
          actionButtons: [
            
          ],
          templateId: 12
        },
        keyNumber: 1,
        body: test,
        title: HiPush,
        actionIdentifier: default,
        key: value1
      }
    }
    ```
 </details>

### CODE REVIEW

File: lib/iterable_flutter.dart [hunk 1]
```diff
@@ -66,13 +68,40 @@ class IterableFlutter {
...
-        _onOpenedNotification?.call(arguments);
+        _onOpenedNotification?.call(argumentsCleaned);
         return "This data from native.....";
       default:
         return "Nothing";
     }
   }
+
+  static Map<String, dynamic> sanitizeMap(
+    Map<dynamic, dynamic> mapDynamic,
```
Comment (cdmunoz) [inline lib/iterable_flutter.dart]: I'd rename this variable with a standard naming convention:
ie: `dynamicsMap` , `dynamicMap`, `schemaMap`, `toMap`

you could pick the option fits the context

File: lib/iterable_flutter.dart [hunk 2]
```diff
@@ -66,13 +68,40 @@ class IterableFlutter {
...
+    bool isAndroidPlatform,
+  ) {
+    final mapHandleDynamic = mapDynamic;
```
Comment (cdmunoz) [inline lib/iterable_flutter.dart]: similar scenario here. Probably `dynamicMapHandler` or the one you consider better

**Review by cdmunoz — COMMENTED** [review]
LGTM 👌🏻 

I left you some small comments regarding variable naming

File: lib/iterable_flutter.dart (hunk 2 above)
Comment (Santi92) [reply lib/iterable_flutter.dart]: Done  ✅

File: lib/iterable_flutter.dart (hunk 1 above)
Comment (Santi92) [reply lib/iterable_flutter.dart]: Ty 👍 
Done ✅

**Review by danielgomezrico — APPROVED** [review]
LGTM

**Review by cdmunoz — APPROVED** [review]
Awesome 🍄  💮

**Pull request merged**